packages/libs/client-examples/README.md

```diff
@@ -1,17 +1,155 @@
-<!-- genericHeader start -->
+## Preparation
 
-# @kadena/client-examples
+In the following examples, we will interact with the `coin` contract. To have
+better type support, we strongly recommend generating the type definition from
+the contract using `@kadena/pactjs-cli`.
 
-Test project to verify pactjs-cli and pactjs-generator
+To install `@kadena/pactjs-cli` as a dev dependency for your project, run the
+following command in the terminal:
 
-<picture>
-  <source srcset="https://raw.githubusercontent.com/kadena-community/kadena.js/main/common/images/Kadena.JS_logo-white.png" media="(prefers-color-scheme: dark)"/>
-  <img src="https://raw.githubusercontent.com/kadena-community/kadena.js/main/common/images/Kadena.JS_logo-black.png" width="200" alt="kadena.js logo" />
-</picture>
+```shell
+npm install @kadena/pactjs-cli --save-dev
+```
 
-<!-- genericHeader end -->
+You can generate type definitions from either a local file or directly from the
+chain.
 
-## Client examples
+### Creating a type definition from a contract deployed on the chain
 
-This project demonstrates the use of the `@kadena/pact-cli` together with
-`@kadena/client` for _smart contracts_.
+```shell
+npx pactjs contract-generate --contract="coin" --api http://104.248.41.186:8080/chainweb/0.0/development/chain/0/pact
+```
+
+### Creating a type definition from a pact file
+
+```shell
+npx pactjs contract-generate --file=./coin.pact --api http://104.248.41.186:8080/chainweb/0.0/development/chain/0/pact
```

Comment:
When I tried this, (with testnet) it gave me an error, the combination of `--file` and `--api` not sure if it's already fixed though (apha.7)

Reply:
This error is expected, you don't need an API when you generate from a file. On a related note I'm wondering where the IP address comes from?

Reply:
that was the devnet url, I have replaced it with the mainnet url.
I also explained when you need --api and when you don't